plugins/woocommerce/src/Internal/DataStores/Orders/OrdersTableMetaQuery.php

```diff
@@ -0,0 +1,566 @@
+<?php
+namespace Automattic\WooCommerce\Internal\DataStores\Orders;
+
+defined( 'ABSPATH' ) || exit;
+
+/**
+ * Class used to implement meta queries for the orders table datastore via {@see OrdersTableQuery}.
+ * Heavily inspired by WordPress' own `WP_Meta_Query` for backwards compatibility reasons.
+ *
+ * Parts of the implementation have been adapted from {@link https://core.trac.wordpress.org/browser/tags/6.0.1/src/wp-includes/class-wp-meta-query.php}.
+ */
+class OrdersTableMetaQuery {
+
+	/**
+	 * List of non-numeric SQL operators used for comparisons in meta queries.
+	 *
+	 * @var array
+	 */
+	private const NON_NUMERIC_OPERATORS = array(
+		'=',
+		'!=',
+		'LIKE',
+		'NOT LIKE',
+		'IN',
+		'NOT IN',
+		'EXISTS',
+		'NOT EXISTS',
+		'RLIKE',
+		'REGEXP',
+		'NOT REGEXP',
+	);
+
+	/**
+	 * List of numeric SQL operators used for comparisons in meta queries.
+	 *
+	 * @var array
+	 */
+	private const NUMERIC_OPERATORS = array(
+		'>',
+		'>=',
+		'<',
+		'<=',
+		'BETWEEN',
+		'NOT BETWEEN',
+
+	);
+
+	/**
+	 * Prefix used when generating aliases for the metadata table.
+	 *
+	 * @var string
+	 */
+	private const ALIAS_PREFIX = 'meta';
+
+	/**
+	 * Sanitized `meta_query`.
+	 *
+	 * @var array
+	 */
+	private $queries = array();
+
+	/**
+	 * JOIN clauses to add to the main SQL query.
+	 *
+	 * @var array
+	 */
+	private $join = array();
+
+	/**
+	 * WHERE clauses to add to the main SQL query.
+	 *
+	 * @var array
+	 */
+	private $where = array();
+
+	/**
+	 * Table aliases in use by the meta query. Used to optimize JOINs when possible.
+	 *
+	 * @var array
+	 */
+	private $table_aliases = array();
+
+	/**
+	 * The meta_query relation.
+	 *
+	 * @var string
+	 */
+	private $relation = 'AND';
+
+	/**
+	 * Constructor.
+	 *
+	 * @param OrdersTableQuery $q The main query being performed.
+	 */
+	public function __construct( OrdersTableQuery $q ) {
+		$meta_query = $q->get( 'meta_query' );
+
+		if ( ! $meta_query ) {
+			return;
+		}
+
+		$this->queries  = $this->sanitize_meta_query( $meta_query );
+		$this->relation = $this->sanitize_relation( $meta_query['relation'] ?? 'AND' );
+
+		$this->meta_table   = $q->get_table_name( 'meta' );
```

Comment:
let's declare this and `orders_table` before using.

Reply:
✔️